lib/crawly/engine.ex

```diff
@@ -33,58 +40,149 @@ defmodule Crawly.Engine do
     GenServer.call(__MODULE__, {:stop_spider, spider_name})
   end
 
-  @spec running_spiders() :: started_spiders()
+  @spec list_spiders() :: [spider_info()]
+  def list_spiders() do
+    GenServer.call(__MODULE__, :list_spiders)
+  end
+
+  @spec running_spiders() :: [spider_info()]
   def running_spiders() do
     GenServer.call(__MODULE__, :running_spiders)
   end
 
+  @spec get_spider(module()) :: spider_info()
+  def get_spider(name) do
+    GenServer.call(__MODULE__, {:get_spider, name})
+  end
+
+  def refresh_spiders() do
+    GenServer.cast(__MODULE__, :refresh_spiders)
+  end
+
   def start_link() do
     GenServer.start_link(__MODULE__, [], name: __MODULE__)
   end
 
   @spec init(any) :: {:ok, __MODULE__.t()}
   def init(_args) do
-    {:ok, %Crawly.Engine{}}
+    spiders = do_refresh_spiders()
+
+    {:ok, %Crawly.Engine{spiders: spiders}}
   end
 
   def handle_call(:running_spiders, _from, state) do
-    {:reply, state.started_spiders, state}
+    state = maybe_refresh_state(state)
+
+    started_spiders =
+      state.spiders
+      |> Enum.filter(fn s -> match?({:started, _pid}, s.status) end)
+
+    {:reply, started_spiders, state}
+  end
+
+  def handle_call({:get_spider, spider_name}, _from, state) do
+    state = maybe_refresh_state(state)
+    spider = do_get_spider_info(state.spiders, spider_name)
+    {:reply, spider, state}
+  end
+
+  def handle_call(:list_spiders, _from, state) do
+    state = maybe_refresh_state(state)
+
+    {:reply, state.spiders, state}
   end
 
   def handle_call({:start_spider, spider_name}, _form, state) do
+    state = maybe_refresh_state(state)
```

Comment:
Let's not refresh the list of spiders implicitly, it makes the code more complex.